collisions.js

```diff
@@ -0,0 +1,128 @@
+#!/usr/bin/env node
+
+// Copyright (c) 2015 Uber Technologies, Inc.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+'use strict';
+
+var AdminClient = require('./lib/admin-client');
+var createTable = require('./lib/table.js').create;
+var detectChecksumMethod = require('./lib/hash-method.js');
+var discover = require('./lib/discover').discover;
+var parseCollisionCommand = require('./parser.js').parseCollisionCommand;
+
+function main() {
+    var command = parseCollisionCommand();
+
+    discover(command.discoveryUri, function onDiscover(err, seeds) {
+        if (err) {
+            console.error('Failed to discover hosts: ', err);
+            process.exit(1);
+            return;
+        }
+
+        printCollisions(command, seeds[0]);
+    });
+}
+
+function detectCollisions(membership, replicaPoints) {
+    var hashFunction = null;
+    var memberHashes = {};
+    var replicaHashes = {};
+
+    var memberCollisionTable = createTable(['hash', 'address', 'collision']);
+    var replicaCollisionTable = createTable(['hash', 'address', '# replica', 'collision', '# replica']);
+
+    function checkMemberCollision(address) {
+        var hash = '' + hashFunction(address);
+        if (!memberHashes[hash]) {
+            memberHashes[hash] = address;
+        } else if (memberHashes[hash] !== address) {
+            //collision!
+            memberCollisionTable.push([hash, address, memberHashes[hash]]);
+        }
+    }
+
+    function checkReplicaCollision(address, replica) {
+        var replicaName = address + replica;
+        var hash = '' + hashFunction(replicaName);
```

Comment:
This function should not be the detected function. The detected function adds a `;` to the `replicaName` when it detects it is `ringpop-go` while the replica points in `go` and `node` are the same:

[ringpop-go implementation](https://github.com/uber/ringpop-go/blob/dev/hashring/hashring.go#L151-L152)
[ringpop-node implementation](https://github.com/uber/ringpop-node/blob/master/lib/ring/index.js#L57-L58)